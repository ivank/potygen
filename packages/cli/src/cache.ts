import { existsSync, readFileSync, writeFileSync, statSync, mkdirSync } from 'fs';
import { dirname } from 'path';
import { createHash } from 'crypto';

export class CachedFileParser<TResult> {
  constructor(public store = new Map<string, TResult>(), public process: (content: string, path: string) => TResult) {}

  public get(path: string): TResult {
    const content = readFileSync(path, 'utf-8');
    const key = createHash('md5').update(content).digest('hex');
    const cachedResult = this.store.get(key);
    if (cachedResult) {
      return cachedResult;
    } else {
      const result = this.process(content, path);
      this.store.set(key, result);
      return result;
    }
  }
}

export class CachedProcessParser<TResult> {
  constructor(public store = new Map<string, TResult>(), public process: (content: string, path: string) => TResult) {}

  public get(path: string): TResult {
    const content = readFileSync(path, 'utf-8');
    const key = createHash('md5').update(content).digest('hex');
    const cachedResult = this.store.get(key);
    if (cachedResult) {
      return cachedResult;
    } else {
      const result = this.process(content, path);
      this.store.set(key, result);
      return result;
    }
  }
}

export class CacheStore {
  public store = new Map<string, number>();

  constructor(public fileName: string, public enabled: boolean = false, public cacheClear: boolean = false) {
    if (this.enabled && !this.cacheClear) {
      if (existsSync(fileName)) {
        this.store = new Map(Object.entries(JSON.parse(readFileSync(fileName, 'utf-8'))));
      }
    }
  }

  public shouldParseFile(path: string) {
    if (!this.enabled) {
      return true;
    } else {
      const mtime = statSync(path).mtime.getTime();
      const currentMtime = this.store.get(path);
      if (currentMtime && mtime === currentMtime) {
        return false;
      }
    }
    return true;
  }

  public cacheFile(path: string) {
    const mtime = statSync(path).mtime.getTime();
    this.store.set(path, mtime);
  }

  public save() {
    if (this.enabled) {
      mkdirSync(dirname(this.fileName), { recursive: true });
      writeFileSync(this.fileName, JSON.stringify(Object.fromEntries(this.store)), 'utf-8');
    }
  }
}
