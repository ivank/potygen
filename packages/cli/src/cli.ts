import { Client } from 'pg';
import { pipeline } from 'stream';
import { promisify } from 'util';
import { SqlRead, QueryLoader } from './traverse';

const asyncPipeline = promisify(pipeline);

const main = async () => {
  const db = new Client({
    database: 'schemes',
    user: 'schemes',
    password: 'dev-pass',
    host: '127.0.0.1',
    port: 5432,
    ssl: false,
  });
  try {
    await db.connect();

    const sqls = new SqlRead('src/**/*.query.ts', '../../../schemes/packages/api');
    const sink = new QueryLoader(
      db,
      '../../../schemes/packages/api',
      '{{root}}/src/queries/__psql-ts__/{{name}}.queries.ts',
    );

    await asyncPipeline(sqls, sink);
  } catch (error) {
    console.log(String(error));
  } finally {
    await db.end();
  }
};

main();
