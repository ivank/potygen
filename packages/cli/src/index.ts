export {
  DataTable,
  DataFunction,
  DataEnum,
  DataView,
  DataComposite,
  LoadedDataTable,
  LoadedDataComposite,
  DataViewRaw,
  DataViewParsed,
  LoadedDataView,
  LoadedDataFunction,
  LoadedDataEnum,
  Data,
  LoadedData,
  LoadedParam,
  LoadedResult,
  LoadedQueryInterface,
  LoadedFunction,
  LoadedComposite,
  LoadedSourceTable,
  LoadedSourceQuery,
  LoadedSourceView,
  LoadedSourceValues,
  LoadedSource,
  LoadedContext,
  TemplateTagQuery,
  ParsedTypescriptFile,
  ParsedSqlFile,
  ParsedFile,
  LoadedTemplateTagQuery,
  LoadedTypescriptFile,
  LoadedSqlFile,
  LoadedFile,
  Logger,
  LoadContext,
  QualifiedName,
} from './types';

export {
  isDataTable,
  isDataFunction,
  isDataEnum,
  isLoadedDataTable,
  isLoadedDataFunction,
  isLoadedDataEnum,
  isLoadedDataView,
  isLoadedDataComposite,
} from './guards';

export { toLoadedContext, loadQueryInterfacesData, toLoadedQueryInterface, loadData, extractDataSources } from './load';

export { Config, ConfigType, FullConfigType, toConfig } from './config';

export { emitLoadedFile, toTypeSource } from './emit';
export { glob } from './glob';
