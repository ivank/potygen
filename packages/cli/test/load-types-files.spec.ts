import { parser } from '@psql-ts/ast';
import { toQueryInterface } from '@psql-ts/query';
import { Client } from 'pg';
import { loadQueryInterface } from '../src/load';
import { sqlFiles, withParserErrors } from './helpers';

let db: Client;

describe('Load Files', () => {
  beforeAll(async () => {
    db = new Client({ database: 'sql-ast', user: 'sql-ast', password: 'dev-pass' });
    await db.connect();
  });

  afterAll(async () => {
    await db.end();
  });

  it.each(sqlFiles())('Should convert complex sql %s', (name, sql) =>
    withParserErrors(async () => {
      const ast = parser(sql);
      const query = toQueryInterface(ast!);
      const { queryInterface } = await loadQueryInterface(db, query);
      expect(queryInterface).toMatchSnapshot(name);
    }),
  );
});
