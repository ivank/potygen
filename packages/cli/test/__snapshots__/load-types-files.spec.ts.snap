// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load Files Should convert complex sql account-levelisation.update.query.sql: account-levelisation.update.query.sql 1`] = `
Object {
  "params": Array [
    Object {
      "name": "isAccepted",
      "type": Object {
        "type": "Boolean",
      },
    },
    Object {
      "name": "id",
      "type": Object {
        "type": "Number",
      },
    },
  ],
  "results": Array [
    Object {
      "name": "id",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "isAccepted",
      "type": Object {
        "type": "Boolean",
      },
    },
  ],
}
`;

exports[`Load Files Should convert complex sql account-levelisation.upsert.query.sql: account-levelisation.upsert.query.sql 1`] = `
Object {
  "params": Array [
    Object {
      "name": "levelisationId",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "start",
      "type": Object {
        "nullable": true,
        "type": "Date",
      },
    },
  ],
  "results": Array [
    Object {
      "name": "id",
      "type": Object {
        "type": "Number",
      },
    },
  ],
}
`;

exports[`Load Files Should convert complex sql account-levelisations.query.sql: account-levelisations.query.sql 1`] = `
Object {
  "params": Array [
    Object {
      "name": "q",
      "type": Object {
        "items": Array [
          Object {
            "literal": "",
            "type": "String",
          },
          Object {
            "type": "String",
          },
        ],
        "type": "UnionConstant",
      },
    },
    Object {
      "name": "resolvedPostlev",
      "type": Object {
        "items": Array [
          Object {
            "literal": "",
            "type": "String",
          },
          Object {
            "type": "Boolean",
          },
        ],
        "type": "UnionConstant",
      },
    },
    Object {
      "name": "state",
      "type": Object {
        "items": Array [
          Object {
            "items": Array [
              Object {
                "literal": "Pending",
                "type": "String",
              },
              Object {
                "literal": "Done",
                "type": "String",
              },
            ],
            "type": "UnionConstant",
          },
          Object {
            "items": Array [
              Object {
                "literal": "Pending",
                "type": "String",
              },
              Object {
                "literal": "Done",
                "type": "String",
              },
            ],
            "nullable": false,
            "type": "UnionConstant",
          },
        ],
        "type": "UnionConstant",
      },
    },
    Object {
      "name": "levelisationId",
      "type": Object {
        "items": Array [
          Object {
            "type": "Number",
          },
          Object {
            "type": "Number",
          },
        ],
        "type": "UnionConstant",
      },
    },
    Object {
      "name": "ids",
      "type": Object {
        "items": Object {
          "type": "Number",
        },
        "type": "ArrayConstant",
      },
    },
    Object {
      "name": "sortField",
      "type": Object {
        "items": Array [
          Object {
            "literal": "totalPayment",
            "type": "String",
          },
          Object {
            "literal": "vatPayment",
            "type": "String",
          },
          Object {
            "literal": "generationPayment",
            "type": "String",
          },
          Object {
            "literal": "exportPayment",
            "type": "String",
          },
          Object {
            "literal": "generationEnergy",
            "type": "String",
          },
          Object {
            "literal": "exportEnergy",
            "type": "String",
          },
          Object {
            "literal": "cfrFitId",
            "type": "String",
          },
          Object {
            "literal": "state",
            "type": "String",
          },
          Object {
            "literal": "isAccepted",
            "type": "String",
          },
          Object {
            "literal": "error",
            "type": "String",
          },
          Object {
            "literal": "exportType",
            "type": "String",
          },
        ],
        "type": "UnionConstant",
      },
    },
    Object {
      "name": "sortOrder",
      "type": Object {
        "items": Array [
          Object {
            "literal": "DESC",
            "type": "String",
          },
          Object {
            "literal": "ASC",
            "type": "String",
          },
        ],
        "type": "UnionConstant",
      },
    },
    Object {
      "name": "limit",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "offset",
      "type": Object {
        "type": "Number",
      },
    },
  ],
  "results": Array [
    Object {
      "name": "id",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "levelisationId",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "accountId",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "installationId",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "state",
      "type": Object {
        "items": Array [
          Object {
            "literal": "Pending",
            "type": "String",
          },
          Object {
            "literal": "Done",
            "type": "String",
          },
        ],
        "nullable": false,
        "type": "UnionConstant",
      },
    },
    Object {
      "name": "isAccepted",
      "type": Object {
        "type": "Boolean",
      },
    },
    Object {
      "name": "generationStartReadOn",
      "type": Object {
        "nullable": true,
        "type": "Date",
      },
    },
    Object {
      "name": "generationStartReadValue",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "generationEndReadOn",
      "type": Object {
        "nullable": true,
        "type": "Date",
      },
    },
    Object {
      "name": "generationEndReadValue",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "generationPercentageSplit",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "generationPayment",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "generationEnergy",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "exportStartReadOn",
      "type": Object {
        "nullable": true,
        "type": "Date",
      },
    },
    Object {
      "name": "exportStartReadValue",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "exportEndReadOn",
      "type": Object {
        "nullable": true,
        "type": "Date",
      },
    },
    Object {
      "name": "exportEndReadValue",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "exportPercentageSplit",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "exportPayment",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "exportEnergy",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "exportType",
      "type": Object {
        "items": Array [
          Object {
            "literal": "Deemed",
            "type": "String",
          },
          Object {
            "literal": "Deemed",
            "type": "String",
          },
          Object {
            "literal": "Metered Export",
            "type": "String",
          },
          Object {
            "literal": "Metered Export",
            "type": "String",
          },
          Object {
            "literal": "Off Grid",
            "type": "String",
          },
          Object {
            "literal": "Off Grid",
            "type": "String",
          },
          Object {
            "literal": "PPA",
            "type": "String",
          },
          Object {
            "literal": "PPA",
            "type": "String",
          },
        ],
        "nullable": true,
        "type": "UnionConstant",
      },
    },
    Object {
      "name": "technologyType",
      "type": Object {
        "items": Array [
          Object {
            "literal": "PV",
            "type": "String",
          },
          Object {
            "literal": "H",
            "type": "String",
          },
          Object {
            "literal": "W",
            "type": "String",
          },
          Object {
            "literal": "AD",
            "type": "String",
          },
          Object {
            "literal": "CHP",
            "type": "String",
          },
        ],
        "nullable": true,
        "type": "UnionConstant",
      },
    },
    Object {
      "name": "totalPayment",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "vatPayment",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "generationPeriods",
      "type": Object {
        "nullable": true,
        "type": "Json",
      },
    },
    Object {
      "name": "exportPeriods",
      "type": Object {
        "nullable": true,
        "type": "Json",
      },
    },
    Object {
      "name": "errorCode",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "errorParams",
      "type": Object {
        "nullable": true,
        "type": "Json",
      },
    },
    Object {
      "name": "cfrFitId",
      "type": Object {
        "type": "String",
      },
    },
    Object {
      "name": "resolvedPostlevId",
      "type": Object {
        "nullable": true,
        "type": "Number",
      },
    },
    Object {
      "name": "isBacsPaymentsSent",
      "type": Object {
        "nullable": true,
        "type": "Boolean",
      },
    },
    Object {
      "name": "isChequePaymentsSent",
      "type": Object {
        "nullable": true,
        "type": "Boolean",
      },
    },
  ],
}
`;

exports[`Load Files Should convert complex sql account-levelisations-bulk-close-accounts.query.sql: account-levelisations-bulk-close-accounts.query.sql 1`] = `
Object {
  "params": Array [
    Object {
      "name": "ids",
      "type": Object {
        "items": Object {
          "type": "Number",
        },
        "type": "ArrayConstant",
      },
    },
  ],
  "results": Array [
    Object {
      "name": "id",
      "type": Object {
        "type": "Number",
      },
    },
  ],
}
`;

exports[`Load Files Should convert complex sql account-levelisations-set-bulk-update.query.sql: account-levelisations-set-bulk-update.query.sql 1`] = `
Object {
  "params": Array [
    Object {
      "name": "ids",
      "type": Object {
        "items": Object {
          "type": "Number",
        },
        "type": "ArrayConstant",
      },
    },
  ],
  "results": Array [
    Object {
      "name": "id",
      "type": Object {
        "type": "Number",
      },
    },
  ],
}
`;

exports[`Load Files Should convert complex sql accounts.list.query.sql: accounts.list.query.sql 1`] = `
Object {
  "params": Array [
    Object {
      "name": "q",
      "type": Object {
        "items": Array [
          Object {
            "literal": "",
            "type": "String",
          },
          Object {
            "type": "String",
          },
          Object {
            "type": "String",
          },
          Object {
            "type": "String",
          },
          Object {
            "type": "String",
          },
          Object {
            "type": "String",
          },
          Object {
            "type": "String",
          },
          Object {
            "nullable": true,
            "type": "String",
          },
          Object {
            "nullable": true,
            "type": "String",
          },
          Object {
            "nullable": true,
            "type": "String",
          },
          Object {
            "nullable": true,
            "type": "String",
          },
          Object {
            "type": "String",
          },
        ],
        "type": "UnionConstant",
      },
    },
    Object {
      "name": "ids",
      "type": Object {
        "items": Object {
          "type": "Number",
        },
        "type": "ArrayConstant",
      },
    },
  ],
  "results": Array [
    Object {
      "name": "total",
      "type": Object {
        "type": "Number",
      },
    },
  ],
}
`;

exports[`Load Files Should convert complex sql accounts.query.sql: accounts.query.sql 1`] = `
Object {
  "params": Array [
    Object {
      "name": "q",
      "type": Object {
        "items": Array [
          Object {
            "literal": "",
            "type": "String",
          },
          Object {
            "type": "String",
          },
          Object {
            "type": "String",
          },
          Object {
            "type": "String",
          },
          Object {
            "type": "String",
          },
          Object {
            "type": "String",
          },
          Object {
            "nullable": true,
            "type": "String",
          },
          Object {
            "nullable": true,
            "type": "String",
          },
          Object {
            "nullable": true,
            "type": "String",
          },
          Object {
            "type": "String",
          },
        ],
        "type": "UnionConstant",
      },
    },
    Object {
      "name": "ids",
      "type": Object {
        "items": Object {
          "type": "Number",
        },
        "type": "ArrayConstant",
      },
    },
    Object {
      "name": "limit",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "offset",
      "type": Object {
        "type": "Number",
      },
    },
  ],
  "results": Array [
    Object {
      "name": "id",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "state",
      "type": Object {
        "items": Array [
          Object {
            "literal": "Active",
            "type": "String",
          },
          Object {
            "literal": "Pending",
            "type": "String",
          },
          Object {
            "literal": "Dispute",
            "type": "String",
          },
          Object {
            "literal": "Closed",
            "type": "String",
          },
        ],
        "nullable": false,
        "type": "UnionConstant",
      },
    },
    Object {
      "name": "startOn",
      "type": Object {
        "nullable": true,
        "type": "Date",
      },
    },
    Object {
      "name": "firstName",
      "type": Object {
        "type": "String",
      },
    },
    Object {
      "name": "lastName",
      "type": Object {
        "type": "String",
      },
    },
    Object {
      "name": "email",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "address",
      "type": Object {
        "type": "String",
      },
    },
    Object {
      "name": "fitIds",
      "type": Object {
        "items": Object {
          "type": "String",
        },
        "type": "ArrayConstant",
      },
    },
  ],
}
`;

exports[`Load Files Should convert complex sql accounts.sql: accounts.sql 1`] = `
Object {
  "params": Array [],
  "results": Array [],
}
`;

exports[`Load Files Should convert complex sql active-meters-wth-reads.query.sql: active-meters-wth-reads.query.sql 1`] = `
Object {
  "params": Array [
    Object {
      "name": "intervalStart",
      "type": Object {
        "items": Array [
          Object {
            "type": "Date",
          },
          Object {
            "type": "Date",
          },
        ],
        "type": "UnionConstant",
      },
    },
    Object {
      "name": "intervalEnd",
      "type": Object {
        "type": "Date",
      },
    },
    Object {
      "name": "accountId",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "meterId",
      "type": Object {
        "items": Array [
          Object {
            "type": "Number",
          },
          Object {
            "type": "Number",
          },
        ],
        "type": "UnionConstant",
      },
    },
  ],
  "results": Array [
    Object {
      "name": "id",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "msn",
      "type": Object {
        "type": "String",
      },
    },
    Object {
      "name": "mpan",
      "type": Object {
        "type": "String",
      },
    },
    Object {
      "name": "make",
      "type": Object {
        "type": "String",
      },
    },
    Object {
      "name": "model",
      "type": Object {
        "type": "String",
      },
    },
    Object {
      "name": "type",
      "type": Object {
        "items": Array [
          Object {
            "literal": "Export",
            "type": "String",
          },
          Object {
            "literal": "Generation",
            "type": "String",
          },
        ],
        "nullable": false,
        "type": "UnionConstant",
      },
    },
    Object {
      "name": "currentPeriodRead",
      "type": Object {
        "items": Array [
          Object {
            "name": "id",
            "type": Object {
              "type": "Number",
            },
          },
          Object {
            "name": "value",
            "type": Object {
              "type": "Number",
            },
          },
          Object {
            "name": "dateOn",
            "type": Object {
              "type": "Date",
            },
          },
          Object {
            "name": "submittedAt",
            "type": Object {
              "nullable": true,
              "type": "Date",
            },
          },
          Object {
            "name": "reason",
            "type": Object {
              "nullable": true,
              "type": "String",
            },
          },
          Object {
            "name": "type",
            "type": Object {
              "items": Array [
                Object {
                  "literal": "Opening",
                  "type": "String",
                },
                Object {
                  "literal": "Closing",
                  "type": "String",
                },
                Object {
                  "literal": "Quarterly",
                  "type": "String",
                },
                Object {
                  "literal": "Meter Verification",
                  "type": "String",
                },
              ],
              "nullable": false,
              "type": "UnionConstant",
            },
          },
        ],
        "type": "ObjectLiteralConstant",
      },
    },
    Object {
      "name": "previousPeriodRead",
      "type": Object {
        "items": Array [
          Object {
            "name": "id",
            "type": Object {
              "type": "Number",
            },
          },
          Object {
            "name": "value",
            "type": Object {
              "type": "Number",
            },
          },
          Object {
            "name": "dateOn",
            "type": Object {
              "type": "Date",
            },
          },
          Object {
            "name": "submittedAt",
            "type": Object {
              "nullable": true,
              "type": "Date",
            },
          },
          Object {
            "name": "reason",
            "type": Object {
              "nullable": true,
              "type": "String",
            },
          },
          Object {
            "name": "type",
            "type": Object {
              "items": Array [
                Object {
                  "literal": "Opening",
                  "type": "String",
                },
                Object {
                  "literal": "Closing",
                  "type": "String",
                },
                Object {
                  "literal": "Quarterly",
                  "type": "String",
                },
                Object {
                  "literal": "Meter Verification",
                  "type": "String",
                },
              ],
              "nullable": false,
              "type": "UnionConstant",
            },
          },
        ],
        "type": "ObjectLiteralConstant",
      },
    },
  ],
}
`;

exports[`Load Files Should convert complex sql bacs.query.sql: bacs.query.sql 1`] = `
Object {
  "params": Array [
    Object {
      "name": "quarter",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "perPage",
      "type": Object {
        "type": "String",
      },
    },
    Object {
      "name": "offset",
      "type": Object {
        "type": "String",
      },
    },
  ],
  "results": Array [
    Object {
      "name": "fitId",
      "type": Object {
        "type": "String",
      },
    },
    Object {
      "name": "name",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "amount",
      "type": Object {
        "type": "String",
      },
    },
    Object {
      "name": "quarter",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "bankDetails",
      "type": Object {
        "type": "String",
      },
    },
    Object {
      "name": "addressLine1",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "addressLine2",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "addressLine3",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "postcode",
      "type": Object {
        "type": "String",
      },
    },
  ],
}
`;

exports[`Load Files Should convert complex sql contracts.sql: contracts.sql 1`] = `
Object {
  "params": Array [],
  "results": Array [],
}
`;

exports[`Load Files Should convert complex sql delete-accounts.query.sql: delete-accounts.query.sql 1`] = `
Object {
  "params": Array [
    Object {
      "name": "type",
      "type": Object {
        "items": Array [
          Object {
            "literal": "FIT",
            "type": "String",
          },
          Object {
            "literal": "SEG",
            "type": "String",
          },
        ],
        "nullable": false,
        "type": "UnionConstant",
      },
    },
    Object {
      "name": "state",
      "type": Object {
        "items": Array [
          Object {
            "literal": "Active",
            "type": "String",
          },
          Object {
            "literal": "Pending",
            "type": "String",
          },
          Object {
            "literal": "Dispute",
            "type": "String",
          },
          Object {
            "literal": "Closed",
            "type": "String",
          },
        ],
        "nullable": false,
        "type": "UnionConstant",
      },
    },
  ],
  "results": Array [
    Object {
      "name": "id",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "sourceSystemId",
      "type": Object {
        "nullable": true,
        "type": "Number",
      },
    },
    Object {
      "name": "customerId",
      "type": Object {
        "type": "Number",
      },
    },
  ],
}
`;

exports[`Load Files Should convert complex sql generation-read-data.query.sql: generation-read-data.query.sql 1`] = `
Object {
  "params": Array [
    Object {
      "name": "installationId",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "exportDateOn",
      "type": Object {
        "type": "Date",
      },
    },
  ],
  "results": Array [
    Object {
      "name": "dateOn",
      "type": Object {
        "type": "Date",
      },
    },
    Object {
      "name": "value",
      "type": Object {
        "type": "Number",
      },
    },
  ],
}
`;

exports[`Load Files Should convert complex sql installation-meter-update.query.sql: installation-meter-update.query.sql 1`] = `
Object {
  "params": Array [
    Object {
      "name": "removalDate",
      "type": Object {
        "nullable": true,
        "type": "Date",
      },
    },
    Object {
      "name": "meterId",
      "type": Object {
        "type": "Number",
      },
    },
  ],
  "results": Array [],
}
`;

exports[`Load Files Should convert complex sql installations.sql: installations.sql 1`] = `
Object {
  "params": Array [],
  "results": Array [],
}
`;

exports[`Load Files Should convert complex sql levelisations.list.query.sql: levelisations.list.query.sql 1`] = `
Object {
  "params": Array [
    Object {
      "name": "q",
      "type": Object {
        "items": Array [
          Object {
            "literal": "",
            "type": "String",
          },
          Object {
            "type": "String",
          },
        ],
        "type": "UnionConstant",
      },
    },
    Object {
      "name": "ids",
      "type": Object {
        "items": Object {
          "type": "Number",
        },
        "type": "ArrayConstant",
      },
    },
    Object {
      "name": "sortField",
      "type": Object {
        "literal": "quarter",
        "type": "String",
      },
    },
    Object {
      "name": "sortOrder",
      "type": Object {
        "items": Array [
          Object {
            "literal": "DESC",
            "type": "String",
          },
          Object {
            "literal": "ASC",
            "type": "String",
          },
        ],
        "type": "UnionConstant",
      },
    },
    Object {
      "name": "limit",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "offset",
      "type": Object {
        "type": "Number",
      },
    },
  ],
  "results": Array [
    Object {
      "name": "id",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "quarter",
      "type": Object {
        "type": "String",
      },
    },
    Object {
      "name": "startOn",
      "type": Object {
        "type": "Date",
      },
    },
    Object {
      "name": "endOn",
      "type": Object {
        "type": "Date",
      },
    },
  ],
}
`;

exports[`Load Files Should convert complex sql meter-reads.sql: meter-reads.sql 1`] = `
Object {
  "params": Array [],
  "results": Array [],
}
`;

exports[`Load Files Should convert complex sql process-batch.query.sql: process-batch.query.sql 1`] = `
Object {
  "params": Array [
    Object {
      "name": "fitIds",
      "type": Object {
        "items": Object {
          "nullable": true,
          "type": "String",
        },
        "type": "ArrayConstant",
      },
    },
  ],
  "results": Array [
    Object {
      "name": "cfrFiTId",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "email",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "contactName",
      "type": Object {
        "type": "String",
      },
    },
    Object {
      "name": "company",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "line1",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "line2",
      "type": Object {
        "type": "String",
      },
    },
    Object {
      "name": "town",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "postcode",
      "type": Object {
        "type": "String",
      },
    },
    Object {
      "name": "county",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "country",
      "type": Object {
        "type": "String",
      },
    },
  ],
}
`;

exports[`Load Files Should convert complex sql rates-by-tarrif-ids.query.sql: rates-by-tarrif-ids.query.sql 1`] = `
Object {
  "params": Array [
    Object {
      "name": "ids",
      "type": Object {
        "items": Object {
          "items": Object {
            "type": "Number",
          },
          "type": "ArrayConstant",
        },
        "type": "ArrayConstant",
      },
    },
  ],
  "results": Array [
    Object {
      "name": "tariffId",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "rate",
      "type": Object {
        "type": "String",
      },
    },
    Object {
      "name": "startOn",
      "type": Object {
        "type": "Date",
      },
    },
    Object {
      "name": "tariffCode",
      "type": Object {
        "type": "String",
      },
    },
    Object {
      "name": "tariffType",
      "type": Object {
        "items": Array [
          Object {
            "literal": "Export",
            "type": "String",
          },
          Object {
            "literal": "Generation",
            "type": "String",
          },
          Object {
            "literal": "SEG",
            "type": "String",
          },
        ],
        "nullable": false,
        "type": "UnionConstant",
      },
    },
    Object {
      "name": "endOn",
      "type": Object {
        "nullable": true,
        "type": "Date",
      },
    },
  ],
}
`;

exports[`Load Files Should convert complex sql read-create.query.sql: read-create.query.sql 1`] = `
Object {
  "params": Array [
    Object {
      "name": "meterId",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "reason",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "type",
      "type": Object {
        "items": Array [
          Object {
            "literal": "Opening",
            "type": "String",
          },
          Object {
            "literal": "Closing",
            "type": "String",
          },
          Object {
            "literal": "Quarterly",
            "type": "String",
          },
          Object {
            "literal": "Meter Verification",
            "type": "String",
          },
        ],
        "nullable": false,
        "type": "UnionConstant",
      },
    },
    Object {
      "name": "dateOn",
      "type": Object {
        "type": "Date",
      },
    },
    Object {
      "name": "submittedAt",
      "type": Object {
        "nullable": true,
        "type": "Date",
      },
    },
    Object {
      "name": "value",
      "type": Object {
        "type": "Number",
      },
    },
  ],
  "results": Array [
    Object {
      "name": "id",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "value",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "meterId",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "submittedAt",
      "type": Object {
        "nullable": true,
        "type": "Date",
      },
    },
  ],
}
`;

exports[`Load Files Should convert complex sql reads-by-account-id.query.sql: reads-by-account-id.query.sql 1`] = `
Object {
  "params": Array [
    Object {
      "name": "id",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "offset",
      "type": Object {
        "type": "String",
      },
    },
    Object {
      "name": "perPage",
      "type": Object {
        "type": "String",
      },
    },
  ],
  "results": Array [
    Object {
      "name": "id",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "msn",
      "type": Object {
        "type": "String",
      },
    },
    Object {
      "name": "meterId",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "dateOn",
      "type": Object {
        "type": "Date",
      },
    },
    Object {
      "name": "value",
      "type": Object {
        "type": "Number",
      },
    },
    Object {
      "name": "type",
      "type": Object {
        "items": Array [
          Object {
            "literal": "Opening",
            "type": "String",
          },
          Object {
            "literal": "Closing",
            "type": "String",
          },
          Object {
            "literal": "Quarterly",
            "type": "String",
          },
          Object {
            "literal": "Meter Verification",
            "type": "String",
          },
        ],
        "nullable": false,
        "type": "UnionConstant",
      },
    },
    Object {
      "name": "reason",
      "type": Object {
        "nullable": true,
        "type": "String",
      },
    },
    Object {
      "name": "createdAt",
      "type": Object {
        "type": "Date",
      },
    },
    Object {
      "name": "updatedAt",
      "type": Object {
        "nullable": true,
        "type": "Date",
      },
    },
    Object {
      "name": "sourceSystemId",
      "type": Object {
        "nullable": true,
        "type": "Number",
      },
    },
    Object {
      "name": "mpan",
      "type": Object {
        "type": "String",
      },
    },
  ],
}
`;

exports[`Load Files Should convert complex sql tariff-rates.sql: tariff-rates.sql 1`] = `
Object {
  "params": Array [],
  "results": Array [],
}
`;
